### ISSUE TITLE
DOC: to_pylist returns a pandas.Timestamp instead of datetime.datetime when the type is timestamp[ns]

### ISSUE THREAD

**Issue by mroeschke** [apache/arrow#34614]
### Describe the enhancement requested

```
In [2]: import pyarrow as pa; from datetime import datetime

In [4]: pa.array([datetime(2023, 1, 1)], type=pa.timestamp("ns")).to_pylist()
Out[4]: [Timestamp('2023-01-01 00:00:00')]
```

From the docs, I assumed `Convert to a list of native Python objects.` means I should always expected a `datetime.datetime`object, but since I suppose it cannot handle nanosecond components this method returns a `pandas.Timestamp` when the unit is `"ns"`

### Component(s)

Python

**Comment by jorisvandenbossche** [comment]
> but since I suppose it cannot handle nanosecond components this method returns a `pandas.Timestamp` when the unit is `"ns"`

That's a correct assumption. I don't think there is anything to do about that? (except for raising an error instead, given that datetime.datetime can't handle nanoseconds)

**Comment by mroeschke** [comment]
Yeah would be good to document this behavior. Or as you mentioned potentially raising. Or warning that nanosecond components would be dropped if storing in a datetime.datetime. I don't have a strong preference either way.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.
